challenges/challenge-weather-app/assets/script.js

```diff
@@ -0,0 +1,139 @@
+  }
+}
+
+// Fetch Unsplash images
+async function fetchImages(query) {
+  const url = `${API_UNSPLASH}?query=${query}&client_id=${UNSPLASH_KEY}`;
+  try {
+    console.log(`Fetching images for: ${query}`);
+    const response = await fetch(url);
+    if (!response.ok) throw new Error("Error fetching images from Unsplash");
+    const data = await response.json();
+    console.log("Unsplash Image Data:", data);
+    return data;
+  } catch (error) {
+    console.error("Error fetching images:", error);
+    throw error;
+  }
+}
+
+// Update the UI with weather and photos
+async function updateWeatherAndPhotos(city) {
+  try {
+    // Fetch weather data
+    const weatherData = await fetchWeather(city);
+    const description = weatherData.weather[0].description;
+    const temperature = weatherData.main.temp;
+    conditionsEl.textContent = `${description}, ${temperature}°C`;
+
+    // Fetch images based on weather description
+    const imageData = await fetchImages(description);
+
+    if (imageData.results.length > 0) {
+      displayImages(imageData.results);
+      updateCredits(imageData.results[0]);
+    } else {
+      console.warn("No images found for the weather description.");
+      photoEl.style.backgroundImage = "";
+      thumbsEl.innerHTML = "";
+      creditUserEl.textContent = "No images available";
+      creditUserEl.href = "#";
+    }
+  } catch (error) {
+    console.error("Error updating weather and photos:", error);
+    alert(error.message);
+    conditionsEl.textContent = "Error: Unable to fetch weather data.";
+  }
+}
+
+// Display images in the main photo section and thumbnails
+function displayImages(images) {
+  if (!images || images.length === 0) return;
+
+  // Set the main photo
+  photoEl.style.backgroundImage = `url(${images[0].urls.regular})`;
+
+  // Clear thumbnails and repopulate
+  thumbsEl.innerHTML = "";
+  images.forEach((image, index) => {
+    const thumb = document.createElement("div");
+    thumb.classList.add("thumb");
+    thumb.style.backgroundImage = `url(${image.urls.thumb})`;
+    thumb.dataset.index = index;
+
+    // Add click event to update main photo and credits
+    thumb.addEventListener("click", () => setMainImage(images, index));
+
+    thumbsEl.appendChild(thumb);
+  });
+
+  // Highlight the first thumbnail as active
+  setActiveThumbnail(0);
+}
+
+// Set the main photo and update active thumbnail
+function setMainImage(images, index) {
+  const image = images[index];
+  photoEl.style.backgroundImage = `url(${image.urls.regular})`;
+  updateCredits(image);
+  setActiveThumbnail(index);
+}
+
+// Update photographer credits
+function updateCredits(image) {
+  creditUserEl.textContent = image.user.name;
+  creditUserEl.href = image.user.links.html;
+  creditPlatformEl.href = "https://unsplash.com";
+}
+
+// Highlight the active thumbnail
+function setActiveThumbnail(activeIndex) {
+  const thumbs = document.querySelectorAll(".thumb");
+  thumbs.forEach((thumb, index) => {
```

Comment:
This works, but it's not using the "state + render" approach that the module discusses - can you talk through what's easier/better about the approach you've taken, and what would be easier/better about the "state + render" approach?